lib/logstash/outputs/tcp.rb

```diff
@@ -56,18 +56,22 @@ class LogStash::Outputs::Tcp < LogStash::Outputs::Base
 
   class Client
 
-    def initialize(socket, logger)
+    def initialize(socket, logger_context)
```

Comment:
Although we are passing in our whole plugin instance (ick), it would be good to document that we intend on using it only for its log helper methods (currently:`log_warn`, possibly:`log_error`) in order to discourage further reach across the boundary.

```suggestion
    ##
    # @param socket [Socket]
    # @param logger_context [#log_warn&#log_error]
    def initialize(socket, logger_context)
```

Reply:
Thanks, added.